pkg/kubelet/volumemanager/cache/desired_state_of_world.go

```diff
@@ -257,15 +257,18 @@ func (dsw *desiredStateOfWorld) AddPodToVolume(
 		var sizeLimit *resource.Quantity
 		if volumeSpec.Volume != nil {
 			if util.IsLocalEphemeralVolume(*volumeSpec.Volume) {
-				_, podLimits := apiv1resource.PodRequestsAndLimits(pod)
-				ephemeralStorageLimit := podLimits[v1.ResourceEphemeralStorage]
-				sizeLimit = resource.NewQuantity(ephemeralStorageLimit.Value(), resource.BinarySI)
 				if volumeSpec.Volume.EmptyDir != nil &&
 					volumeSpec.Volume.EmptyDir.SizeLimit != nil &&
-					volumeSpec.Volume.EmptyDir.SizeLimit.Value() > 0 &&
-					volumeSpec.Volume.EmptyDir.SizeLimit.Value() < sizeLimit.Value() {
+					volumeSpec.Volume.EmptyDir.SizeLimit.Value() > 0 {
 					sizeLimit = resource.NewQuantity(volumeSpec.Volume.EmptyDir.SizeLimit.Value(), resource.BinarySI)
 				}
+				_, podLimits := apiv1resource.PodRequestsAndLimits(pod)
+				ephemeralStorageLimit := podLimits[v1.ResourceEphemeralStorage]
```

Comment:
This code is rearranged more than it needs to be, making it harder to review.  If I'm reading it correctly, the real problem is at the old line 266; it needs to check that the `ephemeralStorageLimit` is not nil, correct?

Reply:
yes, I have add some test case and make it easy to review.